corehq/apps/dashboard/views.py

```diff
@@ -49,22 +49,32 @@ def default_dashboard_url(request, domain):
     return reverse(DomainDashboardView.urlname, args=[domain])
 
 
-@login_and_domain_required
-def dashboard_tile(request, domain, slug):
+def _get_tile(request, slug):
     try:
         tile = [t for t in _get_default_tiles(request) if t.slug == slug][0]
     except IndexError:
         return json_response(
             {'message': _("Tile not found: {}").format(slug)},
             status_code=404,
         )
+    return tile
 
+
+@login_and_domain_required
+def dashboard_tile(request, domain, slug):
+    tile = _get_tile(request, slug)
     current_page = int(request.GET.get('currentPage', 1))
     items_per_page = int(request.GET.get('itemsPerPage', 5))
     items = list(tile.paginator.paginated_items(current_page, items_per_page))
     return json_response({'items': items})
 
 
+@login_and_domain_required
```

Comment:
I think these two views need to be decorated as `@location_safe`.  BTW, what's the difference between `dashboard_tile` and `DomainDashboardView`?  Are tiles sometimes loaded asynchronously, and sometimes not?

Reply:
`DomainDashboardView` is for the entire dashboard, while `dashboard_tile` fetches the items for a single page of a single tile.

Reply:
added `@location_safe`